### PR TITLE
doc(backing image): add ha, scheduling and eviction feature to the doc

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/2856

### CODE REVIEW

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image.md [hunk 1]
```diff
@@ -24,6 +24,14 @@ You can prepare a backing image using four different kinds of data sources.
 - It's recommended to provide the expected checksum during backing image creation.
   Otherwise, Longhorn will consider the checksum of the first file as the correct one. Once there is something wrong with the first file preparation, which then leads to an incorrect checksum as the expected value, this backing image is probably unavailable.
 
+#### Scheduling
+- The backing image file will be first prepared and stored on a random node and a random disk. The file then be duplicated to the disks where replicas are.
+- Users can add `nodeSelector` and `diskSelector` to force BackingImage files to be stored on specific node set and disk set to improve space efficiency.
+- Note that if the `nodeSelector` and `diskSelector` settings are not the same between the volume and the backing image, the replicas will not be able to start if they are scheduled to nodes or disks where backing image can not be placed.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image.md:30]:
```suggestion
- Longhorn first prepares and stores the backing image file on a random node and disk, and then duplicates the file to the disks that are storing the replicas.
- For improved space efficiency, you can add `nodeSelector` and `diskSelector` to force storing of backing image files on a specific set of nodes and disks.
- The replicas will not start if: 1) the `nodeSelector` and `diskSelector` settings for the volume and backing image are not identical, and 2) the replicas are scheduled on nodes or disks that cannot store the backing image.
```

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ You can prepare a backing image using four different kinds of data sources.
...
+
+### Number of copies
+- Users can add `minNumberOfCopies` to ensure there are always multiple backing image files in the cluster to improve availability.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image.md:33]:
```suggestion
- Users can add `minNumberOfCopies` to ensure there are always multiple backing image files in the cluster to improve availability.
```
```suggestion
- You can add `minNumberOfCopies` to ensure that multiple backing image files exist in the cluster.
```

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image.md [hunk 3]
```diff
@@ -147,6 +171,11 @@ Since v1.3.0, users can download existing backing image files to the local via U
   - uploaded from user local env, there is no way to recover it. Users need to delete this backing image then re-create a new one by re-uploading the file.
 - When a node is down or the backing image manager pod on the node is unavailable, all backing image files on the node will become `unknown`. Later on if the node is back and the pod is running, Longhorn will detect then reuse the existing files automatically.
 
+## Backing image eviction
+- Users can manually evict all the backing image files on a node or a disk by setting `Scheduling` to `Disabled` and `Eviction Requested` to `True` on Longhorn UI.
+- If the backing image file is the only file in the cluster, Longhorn will first duplicate it to another disk and then delete the file.
+- Note that if the backing image file can not be duplicated to other disks, Longhorn will not delete the file. Users can update the settings to resolve the situation.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image.md:177]:
```suggestion
- You can manually evict all backing image files from a node or disk by setting `Scheduling` to `Disabled` and `Eviction Requested` to `True` on the Longhorn UI.
- If only one backing image file exists in the cluster, Longhorn first duplicates the file to another disk and then deletes the file.
- If the backing image file cannot be duplicated to other disks, Longhorn does not delete the file. You can update the settings to resolve the issue.
```

**Review by jillian-maroket — COMMENTED** [review]
@ChanYiLin Let me know if you have concerns about the changes.

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image.md (hunk 1 above)
Comment (ChanYiLin) [reply content/docs/1.7.0/advanced-resources/backing-image/backing-image.md:30]: Hi @jillian-maroket 
> - The replicas will not start if: 1) the `nodeSelector` and `diskSelector` settings for the volume and backing image are not identical, and 2) the replicas are scheduled on nodes or disks that cannot store the backing image.

The modification is not correct.
Here is the context
- volume can have multiple replicas
- replicas will be scheduled based on the nodeSelector and diskSelector of the volume
- BackingImage has its own nodeSelector and diskSelector  as well
- if a replica is scheduled to the node where BackingImage can not be scheduled then it won't able to start.

So **it is okay** that **"the `nodeSelector` and `diskSelector` settings for the volume and backing image are not identical"**

Comment (ChanYiLin) [reply content/docs/1.7.0/advanced-resources/backing-image/backing-image.md:30]: But the design has changed, I will provide a new design in the doc, please take a look after, thanks

**Comment by ChanYiLin** [comment]
Hi @jillian-maroket 
I have modified some content, PTAL, thanks!

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image.md [hunk 4]
```diff
@@ -24,6 +24,14 @@ You can prepare a backing image using four different kinds of data sources.
 - It's recommended to provide the expected checksum during backing image creation.
   Otherwise, Longhorn will consider the checksum of the first file as the correct one. Once there is something wrong with the first file preparation, which then leads to an incorrect checksum as the expected value, this backing image is probably unavailable.
 
+#### Scheduling
+- Longhorn first prepares and stores the backing image file on a random node and disk, and then duplicates the file to the disks that are storing the replicas.
+- For improved space efficiency, you can add `nodeSelector` and `diskSelector` to force storing of backing image files on a specific set of nodes and disks.
+- The replica can not be scheduled to the nodes or disks where BackingImage can not be scheduled.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image.md:30]:
```suggestion
- The replicas cannot be scheduled on nodes or disks where the backing image cannot be scheduled.
```

**Review by jillian-maroket — DISMISSED** [review]
@ChanYiLin LGTM overall, will approve now. Just please apply the suggested change to the new line.

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM.

However there are two parts missed.
- A global setting for concurrent BI repliacation
- A gloabl setting for min replication number of BI

**Comment by ChanYiLin** [comment]
Hi @innobead 
Added, the codeSpell errors are not related to this PR
We can fix it later

File: content/docs/1.7.0/references/settings.md [hunk 5]
```diff
@@ -774,6 +781,13 @@ Typically, Longhorn can block the replica starting once the current rebuilding c
 >  - Different from relying on replica starting delay to limit the concurrent rebuilding, if the rebuilding is disabled, replica object replenishment will be directly skipped.
 >  - When the value is 0, the eviction and data locality feature won't work. But this shouldn't have any impact to any current replica rebuild and backup restore.
 
+#### Concurrent Backing Image Replenish Per Node Limit
+
+> Default: `5`
+
+This setting controls how many backing images copy on a node can be replenished simultaneously.
```
Comment (innobead) [inline content/docs/1.7.0/references/settings.md:788]:
```suggestion
This setting controls how many backing image copies on a node can be replenished simultaneously.
```

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
> Hi @innobead Added, the codeSpell errors are not related to this PR We can fix it later

@jillian-maroket there are some code spell errors in the current doc. Please help fix them, so the following PRs will not be bothered with the unrelated errors. Thanks.

**Pull request merged**